### PR TITLE
[23.0] Do not pass dataset keyword parameter into datatype.display_data

### LINKED ISSUE #15676: slugified dataset display broken
from https://sentry.galaxyproject.org/share/issue/889fa180f7274e3bbee66422b45ba049/:

```
AttributeError: 'HistoryDatasetAssociation' object has no attribute 'user'
  File "starlette/applications.py", line 118, in __call__
    await self.middleware_stack(scope, receive, send)
  File "starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "starlette_context/middleware/raw_middleware.py", line 93, in __call__
    await self.app(scope, receive, send_wrapper)
  File "starlette/middleware/base.py", line 109, in __call__
    await response(scope, receive, send)
  File "starlette/responses.py", line 277, in __call__
    await wrap(partial(self.listen_for_disconnect, receive))
  File "anyio/_backends/_asyncio.py", line 662, in __aexit__
    raise exceptions[0]
  File "starlette/responses.py", line 273, in wrap
    await func()
  File "starlette/middleware/base.py", line 134, in stream_response
    return await super().stream_response(send)
  File "starlette/responses.py", line 262, in stream_response
    async for chunk in self.body_iterator:
  File "starlette/middleware/base.py", line 98, in body_stream
    raise app_exc
  File "starlette/middleware/base.py", line 70, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "starlette/routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "starlette/routing.py", line 443, in handle
    await self.app(scope, receive, send)
  File "a2wsgi/wsgi.py", line 155, in __call__
    return await responder(scope, receive, send)
  File "a2wsgi/wsgi.py", line 194, in __call__
    raise self.exc_info[0].with_traceback(
  File "galaxy/web/framework/middleware/error.py", line 165, in __call__
    app_iter = self.application(environ, sr_checker)
  File "/cvmfs/main.galaxyproject.org/venv/lib/python3.8/site-packages/paste/recursive.py", line 85, in __call__
    return self.application(environ, start_response)
  File "galaxy/web/framework/middleware/statsd.py", line 29, in __call__
    req = self.application(environ, start_response)
  File "/cvmfs/main.galaxyproject.org/venv/lib/python3.8/site-packages/paste/httpexceptions.py", line 640, in __call__
    return self.application(environ, start_response)
  File "galaxy/web/framework/base.py", line 166, in __call__
    return self.handle_request(environ, start_response)
  File "galaxy/web/framework/base.py", line 251, in handle_request
    body = method(trans, **kwargs)
  File "galaxy/webapps/galaxy/controllers/dataset.py", line 591, in display_by_username_and_slug
    return trans.fill_template_mako(
  File "galaxy/webapps/base/webapp.py", line 1123, in fill_template_mako
    return template.render(**data)
  File "mako/template.py", line 439, in render
    return runtime._render(self, self.callable_, args, data)
  File "mako/runtime.py", line 874, in _render
    _render_context(
  File "mako/runtime.py", line 916, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "mako/runtime.py", line 943, in _exec_template
    callable_(context, *args, **kwargs)
  File "base.mako.py", line 82, in render_body
    __M_writer(str(next.body()))
  File "display_base.mako.py", line 214, in render_body
    __M_writer(str(self.render_content()))
  File "display_base.mako.py", line 242, in render_render_content
    href_to_user_items = h.url_for( controller='/' + modern_route, action='list_published', xxx=item.user.username)
```

### PR DESCRIPTION
**Pull request by mvdbeek**
https://github.com/galaxyproject/galaxy/pull/15695/commits/8992ec3875ced79919f51706256e865c54535401 fixes https://github.com/galaxyproject/galaxy/issues/15678, https://github.com/galaxyproject/galaxy/pull/15695/commits/d97adf635924476e77958cccea37da77de507450 fixed #15676

Krona generates an html dataset that includes a link with dataset as a query param.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: templates/display_base.mako [hunk 1]
```diff
@@ -101,8 +101,7 @@
         modern_route = modern_route_for_controller(get_controller_name(item))
         item_plural = get_item_plural( item )
         href_to_all_items = h.url_for( controller='/' + modern_route, action='list_published')
-        href_to_user_items = h.url_for( controller='/' + modern_route, action='list_published', xxx=item.user.username)
-        href_to_user_items = href_to_user_items.replace( 'xxx', 'f-username')
+        href_to_user_items = h.url_for( controller='/' + modern_route, action='list_published', **{"f-username": item.user and item.user.username})
```
Comment (jdavcs) [inline templates/display_base.mako:104]: Minor stuff. Do we need to pack and then unpack the `f-username` argument? Maybe just `f-username=item.user and item.user.username`? It's evaluated the same way. (same below)

**Review by jdavcs — APPROVED** [review]

File: templates/display_base.mako (hunk 1 above)
Comment (mvdbeek) [reply templates/display_base.mako:104]: You can't have `-` in argument names:

```
In [2]: dict(f-username=1)
  File "<ipython-input-2-f2b606a950f2>", line 1
    dict(f-username=1)
         ^
SyntaxError: expression cannot contain assignment, perhaps you meant "=="?
```

I think the pack-unpack is a little nicer than what was here before.

**Comment by mvdbeek** [comment]
Whoops, didn't mean to put https://github.com/galaxyproject/galaxy/pull/15695/commits/d97adf635924476e77958cccea37da77de507450 into this PR, but happy to roll with the approval. Thanks @jdavcs!

File: templates/display_base.mako (hunk 1 above)
Comment (jdavcs) [reply templates/display_base.mako:104]: Oh, I see. Sounds good!

**Review by dannon — APPROVED** [review]

**Pull request merged**